jkarenzi/task-master-fe
# Chore: Project setup

### What does this PR do?
- This PR sets up the project using vite

### Description of the task to be completed
- [x] Set up react, typescript project using vite
- [x] Set up testing environment using vitest, @testing-library/react and @testing--library/jest-dom
- [x] Set up CI/CD using github actions
- [x] Set up redux for global state management
- [x] Set up linting and formatting (also set up pre commit hooks using husky)
- [x] Install base dependencies. e,g react-router-dom, redux, and axios
- [x] Dockerize the project

### How can this be manually tested?
- Accessing the home page should display some text
- Tests can be ran using npm run test
- linting and formatting can be run using npm run lint and npm run format commands respectively

### What are the relevant pivotal trackers/story id?
- #1
